source/Ship.cpp

```diff
@@ -4794,3 +4844,10 @@ double Ship::CalculateDeterrence() const
 		}
 	return tempDeterrence;
 }
+
+
+
+bool Ship::BayContains(const Bay &bay, const string &category) const
+{
+	return (bay.bayType && bay.bayType->Contains(category)) || (!bay.bayType && bay.name == category);
```

Comment:
```suggestion
	if(bay.bayType)
		return bay.bayType->Contains(category);
	else
		return bay.name == category;
```
Should be more readable this way